### ISSUE TITLE
Add transition dsl for internal transitions

### ISSUE THREAD

**Issue by erikzenker** [erikzenker/hsm#66]
Such as:
`hsm::event<e1>`
`hsm::event<e1> [guard]`
`hsm::event<e1> / action`
`hsm::event<e1> [guard] / action`

**Comment by erikzenker** [comment]
After first investigations of how to implement this feature it turned out that the current design of internal transition does not make it easy to implement. Currently the internal transition table looks the following:
```c++

    constexpr auto make_internal_transition_table()
    {
        return hsm::transition_table(
            hsm::transition(hsm::event<e1> {}, g1, a1),
            hsm::transition(hsm::event<e2> {}, g1, a2)
        );
    }
```
So the expected type of the function is a tuple of tuples. This type is currently not returned by the `[]` and `/` operator of `hsm::event`

**Comment by erikzenker** [comment]
The transition dsl for internal transitions was merged with #93 and looks as follows:
```c++
+ (hsm::event<e1>)
+ (hsm::event<e1> [guard])
+ (hsm::event<e1> / action)
+ (hsm::event<e1> [guard] / action)
```

I updated the integration test for internal transitions
